test/links/connection/embed_id_test.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+class Chainer::Links::Connection::EmbedIDTest < Test::Unit::TestCase
+  data = {
+    test1: {x_data: [0, 1, 0], ignore_label: nil},
+    test2: {x_data: [[0, 1, 0], [1, 0, 1]], ignore_label: nil},
+    test3: {x_data: [0, 1, -1], ignore_label: -1},
+    test4: {x_data: [[0, 1, -1], [-1, 0, 1]], ignore_label: -1},
+  }
+
+  def _setup(data)
+    @link = Chainer::Links::Connection::EmbedID.new(3, 2, ignore_label: data[:ignore_label])
+    @link.cleargrads
+
+    @w = @link.w.data.copy()
+    @x = Numo::Int32[*data[:x_data]]
```

Comment:
Use `xm` which is defined at `run_test.rb`.
Set `RED_CHAINER_GPU` environment variable to use Cumo.

See https://github.com/red-data-tools/red-chainer#run-tests-with-cumo about how to run tests with Cumo.